pkg/config/user_config.go

```diff
@@ -123,6 +123,8 @@ type GuiConfig struct {
 	NerdFontsVersion string `yaml:"nerdFontsVersion" jsonschema:"enum=2,enum=3,enum="`
 	// If true (default), file icons are shown in the file views. Only relevant if NerdFontsVersion is not empty.
 	ShowFileIcons bool `yaml:"showFileIcons"`
+	// Length of commit ID/ref (hash) in commits view.
+	CommitHashLength int `yaml:"commitHashLength" jsonschema:"minimum=1,maximum=12"`
```

Comment:
I initially set it 12 arbitrarily to avoid out of range index errors but that's redundant because of the way I wrote the code to display it. I also wasn't sure if it would mess up anything elsewhere because I'm not very familiar with the repo.

I changed it to maximum 40 here 3816cb90189fdffb5522c069979fe00dd8dd74b1 because I think having it in the schema would be nice to let people know how long hashes are (I didn't know)!

Reply:
Hm, but this will change in the future. The git folks are working on replacing the 40-characters SHA1 with another hash algorithm that has longer hashes. This won't happen very soon, but some day, and the 40-character maximum will be as arbitrary then as the 12 would be today.

I guess in the light of this it's still best to leave out maximum, and allow users to set arbitrarily large values. And then it would be better to clamp out-of-range values to the length of the hash instead of falling back to ShortHash as you do now.

Reply:
Great points!
I removed the schema maximum and the ShortHash call.
I also set the minimum to 0 and made up for the fact that without NF icons you couldn't see the status color by adding a display character (default is ●).

https://github.com/jesseduffield/lazygit/blob/8210a79ecd8da642b4dcccf969234d7ef9a2f1da/pkg/config/user_config.go#L126-L131

https://github.com/jesseduffield/lazygit/blob/8210a79ecd8da642b4dcccf969234d7ef9a2f1da/pkg/gui/presentation/commits.go#L317-L325